test/test_cuda.py

```diff
@@ -1678,36 +1677,76 @@ def test_events(self):
         self.assertGreater(start_event.elapsed_time(event), 0)
 
     @staticmethod
-    def _stream_synchronize(spin_time=50000000):
+    def _stream_synchronize(self, spin_time=50000000):
         s = torch.cuda.current_stream()
+        e_tik = torch.cuda.Event(enable_timing=True)
+        e_tok = torch.cuda.Event(enable_timing=True)
+
+        torch.cuda._sleep(1)
```

Comment:
Out of curiosity, how is this implemented?

Reply:
Also, why is this necessary?

Reply:
> Also, why is this necessary?

I thought event time was aligned to the the completion of the last computation/communication operation. So the `_sleep(1)` is trying to align the starting event. But, it turns out I am wrong. Event marks the record time, as long as the stream is not busy. I will remove it. Thanks!